backend/layers/business/business.py

```diff
@@ -0,0 +1,190 @@
+from ast import Str
+from dataclasses import dataclass
+from typing import Iterable, Optional
+from backend.corpora.common.providers.crossref_provider import CrossrefProvider
+
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetStatus, DatasetVersion
+from backend.layers.persistence.persistence import DatabaseProviderInterface
+from backend.layers.thirdparty.crossref_provider import CrossrefProviderInterface
+from backend.layers.thirdparty.step_function_provider import StepFunctionProviderInterface
+
+
+@dataclass
+class CollectionQueryFilter:
+    from_date: Optional[str]
+    to_date: Optional[str]
+    visibility: Optional[str] # Use an enum
+    owner: Optional[str]
+    # TODO: add list of fields to be returned (if needed)
+
+@dataclass
+class UserInfo:
+    user_id: str
+    token: str
+
+
+class BusinessLogicInterface:
+
+    # Get_collections
```

Comment:
this should be a docstring under get_collections

```python
def get_collections():
    """ 
        Replaces get_collections_list and get_collections_index
        Accepts a CollectionFilter class (or kwargs) with:
        Date bounds
        Visibility
        Ownership
        List of fields to be returned
        Accepts an UserInfo object
        Returns a list of dictionaries that only includes the selected fields for each collection
        It should NOT add any information that is required by the current API for compatibility reasons (e.g. access_write). These will be delegated the upper layer
        It should NOT do any operation that is required by Curation API assumptions (e.g. remove None values)
"""
```

Reply:
reading these comments and how mentions the soon to be deprecaited functions, it reads more like notes for future implementation not actual documentation. Is that how I should be interpretting it?